zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/ZeebeRocksDBMetricExporter.java

```diff
@@ -92,10 +93,13 @@ public void exportMetrics() {
   private void exportMetrics(final RocksDBMetric[] metrics) {
     for (final RocksDBMetric metric : metrics) {
       try {
-        database
-            .getProperty(metric.getPropertyName())
-            .map(Double::parseDouble)
-            .ifPresent(value -> metric.exportValue(partition, value));
+        final var database = databaseSupplier.get();
```

Comment:
:thought_balloon: Maybe move this outside the for loop. But I am not sure myself. This has more lookup but is more robust if e.g. the database closes while iterating over the metrics.